admin/collection_providers/views.py

```diff
@@ -21,6 +21,17 @@
 from admin.providers.views import AddAdminOrModerator, RemoveAdminsAndModerators
 
 
+def process_collection_choices(self, choices_name, form):
```

Comment:
Need to pass the collection (and not expect self) if doing it this way, by convention, I would also make this a "protected" method, i.e. name it something like `_process_collection_metadata_choices` with the `_` in front